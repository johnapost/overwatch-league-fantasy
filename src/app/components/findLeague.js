// @flow

import React from "react";
import { compose } from "redux";
import { connect } from "react-redux";
import { Modal, Button, Divider, Form, Input } from "antd";

import type { StoreState } from "../shared/makeStore";

type Props = {
  displayName: string
};

export const FindLeagueComponent = ({ displayName }: Props) => (
  <Modal
    title={`Welcome, ${displayName}`}
    visible
    footer={null}
    keyboard={false}
    autoFocusButton={null}
    closable={false}
  >
    <p>Join an existing league or create your own!</p>
    <Form layout="inline">
      <Form.Item>
        <Input placeholder="Invite code" disabled />
      </Form.Item>
      <Form.Item>
        <Button type="primary" disabled>
          Join someone&rsquo;s league
        </Button>
      </Form.Item>
    </Form>
    <Divider />
    <Form layout="inline">
      <Form.Item>
        <Input placeholder="League name" />
      </Form.Item>
      <Form.Item>
        <Button>Create a league</Button>
      </Form.Item>
    </Form>
    <style jsx>{`
      p {
        margin: 10px 0 30px;
      }
    `}</style>
  </Modal>
);

export const mapStateToProps = ({ user: { displayName } }: StoreState) => ({
  displayName
});

export default compose(connect(mapStateToProps))(FindLeagueComponent);
