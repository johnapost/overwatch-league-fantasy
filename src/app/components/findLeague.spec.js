import React from "react";
import { shallow } from "enzyme";
import { Modal } from "antd";
import { FindLeagueComponent, mapStateToProps } from "./findLeague";

const props = {
  displayName: "John again"
};

describe("FindLeague", () => {
  describe("component", () => {
    const wrapper = shallow(<FindLeagueComponent {...props} />);

    it("should render user displayName", () => {
      expect(wrapper.find(Modal).prop("title")).toMatch(props.displayName);
    });
  });

  describe("mapStateToProps", () => {
    it("should assign displayName", () => {
      expect(
        mapStateToProps({ user: { displayName: "user" } })
      ).toMatchSnapshot();
    });
  });
});
