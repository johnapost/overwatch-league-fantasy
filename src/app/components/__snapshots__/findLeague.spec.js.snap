// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FindLeague mapStateToProps should assign displayName 1`] = `
Object {
  "displayName": "user",
}
`;
